### PR TITLE
Add object support to concat custom function

### PR DESCRIPTION
**Pull request by sordina**
This PR adds support for concatenating objects to the concat custom function.

Kriti:

```kriti
{{ concat([{"a":1, "b":2}, {"b":3, "c":4}]) }}
```

Output:

```json
{"a":1, "b":3, "c":4}
```

### CODE REVIEW

File: src/Kriti/CustomFunctions.hs [hunk 1]
```diff
@@ -134,12 +135,14 @@ removeNullsF = parserToFunc $ J.withArray "Array" \a -> do
...
+-- `[{"a":1, "b":2}, {"b":3, "c":4}]` -> `{"a":1, "b":3, "c":4}`
 concatF :: KritiFunc
 concatF = parserToFunc $ J.withArray "Array" \as -> do
   let l = V.toList as
-      a = J.Array . V.concat <$> traverse (J.withArray "Nested Array" pure) l
-      s = J.String . T.concat <$> traverse (J.withText "Nested String" pure) l
-  a <|> s
+      a = J.Array . fold <$> traverse (J.withArray "Nested Array" pure) l
+      s = J.String . fold <$> traverse (J.withText "Nested String" pure) l
+      o = J.Object . fold . reverse <$> traverse (J.withObject "Nested Object" pure) l
```
Comment (solomon-b) [inline src/Kriti/CustomFunctions.hs:144]: You could probably use `foldM` instead of `traverse` here to avoid the `fold . reverse`.

**Review by solomon-b — APPROVED** [review]
Looks good! I left one comment about trying out `foldM` but its not blocking.

File: src/Kriti/CustomFunctions.hs (hunk 1 above)
Comment (sordina) [reply src/Kriti/CustomFunctions.hs:144]: I think foldM takes a folding function so I'd prefer to use a version without that requirement if possible.

Comment (sordina) [reply src/Kriti/CustomFunctions.hs:144]: The type is actually pretty gnarly:

```haskell
foo :: (Monoid (t1 b), Traversable t1, Applicative t2, Foldable t2) => (a -> t2 b) -> t1 a -> t1 b
foo f = fold . traverse f
```

Plus this doesn't handle the 'reverse' so I'd need to use that fold reversal newtype.

Comment (sordina) [reply src/Kriti/CustomFunctions.hs:144]: Maybe I'm missing something that would make this work though.

**Comment by sordina** [comment]
I'll merge as is for now, but would like to get your thoughts about the fold idea when you're free.

File: src/Kriti/CustomFunctions.hs (hunk 1 above)
Comment (solomon-b) [reply src/Kriti/CustomFunctions.hs:144]: Sorry `foldM` was wrong. I think you could do something like:
```
bar :: [J.Value] -> J.Parser J.Value
bar l = foldl' (\p val -> fmap J.Object (J.withObject "Nested Object" pure val) <> p) mempty l
```
And then you don't need to traverse the structure multiple times to build the parsers, reverse them, then fold them together.

**Pull request merged**